pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenvNoCC, lib, writeTextDir, php }:
+
+let
+  buildPhpProjectOverride = finalAttrs: previousAttrs:
+
+  let
+    phpDrv = finalAttrs.php or php;
+    composer = finalAttrs.composer or phpDrv.packages.composer;
+
+    prefix = builtins.replaceStrings ["-"] [""] finalAttrs.pname;
+
+    # See https://github.com/composer/composer/pull/9212
+    apcuAutoloaderPrefixArg = "--apcu-autoloader-prefix=${prefix}";
+
+    # See https://getcomposer.org/doc/03-cli.md#composer-home-config-json
+    composerHome = let
+      content.config.autoloader-suffix = prefix;
+    in
+      writeTextDir "/config.json" (builtins.toJSON content);
+
+    vendor = stdenvNoCC.mkDerivation (finalVendorAttrs: {
+      inherit apcuAutoloaderPrefixArg;
+      inherit (finalAttrs) version src;
+      pname = "${finalAttrs.pname}-vendor";
+
+      # See https://github.com/NixOS/nix/issues/6660
+      dontPatchShebangs = true;
+
+      nativeBuildInputs = [
+        composer
+        php.composerHooks.composerSetupHook
+      ];
+
+      composerLock = finalAttrs.composerLock or null;
+
+      # Should we keep these empty phases?
+      configurePhase = ''
+        runHook preConfigure
+
+        runHook postConfigure
+      '';
+
+      buildPhase = ''
+        runHook preBuild
+
+        runHook postBuild
+      '';
+
+      installPhase = ''
+        runHook preInstall
+
+        runHook postInstall
+      '';
+
+      strictDeps = true;
+
+      outputHashMode = "recursive";
+      outputHashAlgo = if (finalAttrs ? vendorHash && finalAttrs.vendorHash != "") then null else "sha256";
+      outputHash = finalAttrs.vendorHash or "";
+    });
```

Comment:
Can you use `buildComposerHooks { inherit (finalAttrs) vendorHash php composer composerLock; }` here?

Reply:
yes. I merged this in the original PR, let's continue there.

Reply:
I pushed the changes, but I couldn't implement it exactly as you wanted. Let me know if it's ok.  Relevant commit: https://github.com/NixOS/nixpkgs/pull/225401/commits/4a484ee2d49aa7db04f37c2eb24474834bcc0a72